payjoin/src/core/receive/v2/mod.rs

```diff
@@ -377,26 +366,28 @@ impl Receiver<Initialized> {
         let proposal = match self.inner_process_res(body, context) {
             Ok(proposal) => proposal,
             Err(e) => match e {
-                // Implementation errors should be unreachable
+                // Fatal V2 errors result in immediate session closure because it is infeasible
+                // to reply to the sender
                 ProtocolError::V2(ref session_error) => match session_error {
                     SessionError(InternalSessionError::DirectoryResponse(directory_error)) =>
                         if directory_error.is_fatal() {
                             return MaybeFatalTransitionWithNoResults::fatal(
-                                SessionEvent::SessionInvalid(e.to_string(), None),
+                                SessionEvent::Closed(SessionOutcome::Failure),
                                 e,
                             );
                         } else {
                             return MaybeFatalTransitionWithNoResults::transient(e);
                         },
                     _ =>
                         return MaybeFatalTransitionWithNoResults::fatal(
-                            SessionEvent::SessionInvalid(session_error.to_string(), None),
+                            SessionEvent::Closed(SessionOutcome::Failure),
                             e,
                         ),
                 },
+                // Replyable errors should not close the session until a reply was attempted
                 _ =>
                     return MaybeFatalTransitionWithNoResults::fatal(
-                        SessionEvent::SessionInvalid(e.to_string(), None),
+                        SessionEvent::HasReplyableError((&e).into()),
```

Comment:
Would persisting this event progress to the next type state? In this case being `HasReplyError`? OR would the application first have to replay the log?